designs/autoscaler.md

```diff
@@ -0,0 +1,515 @@
+
+### Cluster Autoscaler Controller
+
+Cluster Autoscaler is a tool that automatically adjusts the size of the Kubernetes cluster. Specifically, it can:
+
+* increase the size of the cluster when there are pending pods due to insufficient resources.
+* decrease the size of the cluster when some nodes have low utilization for a significant amount of time.
+
+Just like any other application, Cluster Autoscaler can be installed as a `Deployment` in any Kubernetes cluster. In order to connect Cluster Autoscaler to Cluster API managed EKS Anywhere cluster, a user needs to configure Cluster Autoscaler to use `clusterapi` as the cloud provider, and provide the path to the kubeconfig for the target cluster to autoscale.
+
+Currently, it is impossible to have one autoscaler controller to watch multiple workload clusters with different kubeconfig at the same time, as there is no way currently to separate the pods between the workload clusters autoscaler is watching. The recommended method is to run multiple autoscalers in the management cluster with different options for how they discover the scalable resource sets.
+
+Cluster Autoscaler provides the flexibility to enable autoscaling, whether it is within the cluster, or for a remote workload cluster. Cluster Autoscaler can be:
+
+* deployed in the management cluster to autoscale the management cluster itself
+* deployed in the management cluster to autoscale a remote workload cluster
+* deployed in the workload cluster to autoscale the workload cluster itself
+
+Since EKS Anywhere follows a standard management/workload clusters setup where all the CAPI machine objects are defined in the management cluster, we choose to **deploy autoscalers in the management cluster as well — to watch and manage management cluster, and all the remote workload clusters, as default option**. In this way, we have a centralized place — management cluster, to deploy and manage all the autoscaler controllers for workload clusters. 
+
+However, it is not uncommon that the management cluster size defined by the user is small, but the workload clusters are rather big, then it makes more sense to distribute the workload and run the autoscaler in each workload cluster to reduce resource usage in one management cluster. This configuration is also supported by us. Though this approach requires extra steps to create the management cluster’s `kubeconfig` in the workload cluster, comparing to the first approach where each workload cluster’s kubeconfig by default exists in the management cluster. Having management cluster kubeconfig in workload cluster also draws security concerns. 
```

Comment:
Are we able to elaborate a little on the security concerns of external controllers given its a valid deployment model? It'll be useful for docs given we're saying its supported.

Reply:
yup, updated